<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:context="http://www.springframework.org/schema/context"
	xsi:schemaLocation="
        http://www.springframework.org/schema/beans
        http://www.springframework.org/schema/beans/spring-beans.xsd
        http://www.springframework.org/schema/context
        http://www.springframework.org/schema/context/spring-context.xsd">

	<bean id="backEndErrorFactory" class="com.qsocialnow.rest.response.BackEndErrorFactory">
		<property name="exceptionMapping">
			<map>
				<entry key="Unknown">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="500" />
						<property name="code" value="1" />
						<property name="message" value="Hubo un error inesperado" />
					</bean>
				</entry>
				<entry key="org.springframework.dao.DataIntegrityViolationException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="500" />
						<property name="code" value="2" />
						<property name="message"
							value="Hubo una violación de integridad de datos" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.ChangePasswordException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="500" />
						<property name="code" value="3" />
						<property name="message" value="Hubo un error al cambiar el password" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.AuthenticationException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="401" />
						<property name="code" value="4" />
						<property name="message" value="Usuario o password inválido" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.RefreshTokenException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="500" />
						<property name="code" value="5" />
						<property name="message"
							value="Hubo un error al querer refrescar el token" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.ForbiddenException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="403" />
						<property name="code" value="6" />
						<property name="message"
							value="El usuario ingresado no está autorizado para acceder al sistema" />
					</bean>
				</entry>
				<entry key="org.springframework.transaction.TransactionSystemException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="500" />
						<property name="code" value="7" />
						<property name="message"
							value="Hubo una error al hacer commit de los datos. Probablemente se deba a una violación de integridad. Consulte el log para más detalle." />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.TokenExpiredException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="401" />
						<property name="code" value="8" />
						<property name="message"
							value="Token Expirado. Puede renovarlo invocando el servicio de /refresh" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.UnauthorizedException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="401" />
						<property name="code" value="9" />
						<property name="message" value="No autorizado!" />
					</bean>
				</entry>
				<entry key="com.fasterxml.jackson.core.JsonProcessingException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="400" />
						<property name="code" value="10" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.NotFoundException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="404" />
						<property name="code" value="11" />
						<property name="message" value="No encontrado!" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.ValidationException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="400" />
						<property name="code" value="12" />
					</bean>
				</entry>
				<entry key="com.qsocialnow.common.exception.ServiceException">
					<bean class="com.qsocialnow.rest.response.BackEndError">
						<property name="httpCode" value="500" />
						<property name="code" value="13" />
						<property name="message" value="Una excepción genérica de servicio con parametro = {0}" />
					</bean>
				</entry>				
			</map>
		</property>
	</bean>

	<bean id="restResponseHandler" class="com.qsocialnow.rest.response.RestResponseHandler" />

</beans>